### PR TITLE
fix(node): v8 deserialize accepts Buffer, TypedArray and DataView as input

### PR DESCRIPTION
**Pull request by peterhirn**
fix(node): v8 deserialize accepts Buffer, TypedArray and DataView as input

see https://nodejs.org/api/v8.html#v8deserializebuffer

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/api/v8.html#v8deserializebuffer
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:

- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Review by jakebailey — CHANGES_REQUESTED** [review]
You should be adding tests, and also updating older versions of the Node types if this also applies to them.

File: types/node/v8.d.ts [hunk 1]
```diff
@@ -445,7 +445,7 @@ declare module "v8" {
      * @since v8.0.0
      * @param buffer A buffer returned by {@link serialize}.
      */
-    function deserialize(buffer: NodeJS.TypedArray): any;
+    function deserialize(buffer: Buffer | NodeJS.TypedArray | DataView): any;
```
Comment (Renegade334) [inline types/node/v8.d.ts:448]:
```suggestion
    function deserialize(buffer: NodeJS.ArrayBufferView): any;
```
`NodeJS.ArrayBufferView` is the canonical alias.

**Review by Renegade334 — CHANGES_REQUESTED** [review]

**Comment by peterhirn** [comment]
> You should be adding tests, and also updating older versions of the Node types if this also applies to them.

Looks like the functions signature is stable since Node v10: https://nodejs.org/docs/latest-v10.x/api/v8.html#v8_v8_deserialize_buffer

Please take a look at my tests before I copy the fix to v16/18/20.

**Comment by Renegade334** [comment]
> Please take a look at my tests before I copy the fix to v16/18/20.

`ArrayBuffer` and `string` are "buffery" so definitely good to test; `number` is probably unnecessary.

**Review by jakebailey — APPROVED** [review]

**Comment by peterhirn** [comment]
Ready to merge

**Pull request merged**